### PR TITLE
Run e2e,integration tests against supported kubernetes versions

### PR DESCRIPTION
**Pull request by clux**
As part of Kubernetes version policies for #684. 
Actual proposed policies are in kube-rs/website#19 and is mostly in line with our current advice.

This is larger overhaul for the CI situation for e2e tests to at least guarantee that the various Kubernetes versions work:

Screenshot from 2022-06-06 21-43-28

Main stuff done:
- [x] factored out integration tests from coverage job
- [x] running integration tests against [mk8sv, latest] (found one timing issue)
- [x] extend e2e incluster tests to run both ssl stacks
- [x] add e2e compile test to verify mk8sv and latest feature sets
- [x] extend integration tests to run against different cluster versions (but same feature set)
- [x] update automation to bump mk8sv and badge (just bump-k8s also bumps minimum now)
- [x] improve minimum verification from badges (both on MSRV and MK8SV)
- [x] general ci workflow improvement on naming and organisation (lint factored into its own, lowercase names)
- [x] update required statuses after review

**k3s complication:** k3s is lagging behind k8s; [pre-releases](https://github.com/k3s-io/k3s/releases) available 1w later, but 1mo later no stable 1.24.. => pinning only minimum supported and running against latest => can only really test against minimum and max available.

Verification strategy:
- run integration tests against `v1_19` (but without changing cluster version)
- compile `e2e/boot.rs` against both latest k8s-openapi feature and our msrv (currently `v1_19` feature)

Thus the compile check avoids the error where we accidentally publish abstractions on apis that k8s-openapi have removed, and the integration tests run with a later k8s-openapi no matter what (because it's super hard to untangle that, and also not super necessary).

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (kazk) [inline .github/workflows/ci.yml:125]: Testing against older Kubernetes version is a good start.

> Turns out we can define a minimum support and run simple tests against stable apis without changing k8s-openapi feature version at a start (because our tests don't rely on deprecated/alpha apis). 

Our tests don't use deprecated/alpha APIs currently, but don't we need to change `k8s-openapi`'s feature to prevent merging accidental changes that cause compile errors like the original one in #684?

Maybe we can just add a separate job that simply checks if `k8s-openapi/v1_XX` compiles.

Comment (clux) [reply .github/workflows/ci.yml:125]: Good point, we do want to prevent that class of failures.

I might try to make an `e2e` test for this because it's hard to just compile `k8s-openapi/v1_XX` against `kube` without exposing tons of useless features on all our crates. In `e2e` we can instead have a single bin that can uses an older `k8s-openapi` without having to interact with our dev-deps.

Comment (clux) [reply .github/workflows/ci.yml:141]: Ok, this is now done. We have an `e2e/boot.rs` test that does this, and some general improvements to ci has been done in general to make things nicer.

**Comment by clux** [comment]
Am fairly happy with the way this is now. It matches the policy in the upstream website PR, and we have better e2e tests to at least compile time guarantee we are not building on top of removed apis. Minor improvements here and there as outlined in edited main post. Feedback welcome.

File: kube-runtime/src/wait.rs
Comment (kazk) [inline kube-runtime/src/wait.rs:220]: Should this be in a separate PR? It won't be listed in the changelog.

**Comment by kazk** [comment]
Not sure why this is stuck.

By the way, I just enabled https://github.blog/changelog/2022-06-08-admins-can-require-sign-off-on-web-based-commits at org level, which will prevent the DCO issue when committing suggested changes.

**Comment by clux** [comment]
> Not sure why this is stuck.

It's not actually stuck. This PR has changed jobs + job names, so I need to change required statuses before merging (hence why the required ones are not reported).

> By the way, I just enabled https://github.blog/changelog/2022-06-08-admins-can-require-sign-off-on-web-based-commits at org level, which will prevent the DCO issue when committing suggested changes.

Ah, that sounds great. Thanks!

File: kube-runtime/src/wait.rs
Comment (clux) [reply kube-runtime/src/wait.rs:220]: I can factor it into its own PR.

Comment (clux) [reply kube-runtime/src/wait.rs:220]: https://github.com/kube-rs/kube-rs/pull/935. Let's do that one first.

**Review by kazk — APPROVED** [review]

**Pull request merged**
